### PR TITLE
DocumentationStyleBear: Use DocBaseClass

### PR DESCRIPTION
**Pull request by damngamerz**
DocumentationStyleBear uses DocBaseClass.
Closes https://github.com/coala/coala-bears/issues/1927

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

**Comment by damngamerz** [comment]
This should be merged after https://github.com/coala/coala/pull/4465
Till then status blocked.

File: bears/documentation/DocumentationStyleBear.py [hunk 1]
```diff
@@ -20,6 +21,68 @@ class DocumentationStyleBear(LocalBear):
...
+        # Assuming that the first element is always the only main
+        # description.
+        metadata = iter(parsed)
+
+        main_description = next(metadata)
+
+        if main_description.desc == '\n' and not allow_missing_func_desc:
+            # Triple quoted string literals doesn't look good. It breaks
+            # the line of flow. Hence we use dedent.
+            self.warning_desc = dedent("""\
```
Comment (NiklasMM) [inline bears/documentation/DocumentationStyleBear.py]: Yes! :tada:

File: bears/documentation/DocumentationStyleBear.py [hunk 2]
```diff
@@ -46,75 +109,29 @@ def run(self, filename, file, language: str,
...
-
-                # We need to update old comment positions, as `assemble()`
-                # prepends indentation for first line.
-                old_range = TextRange.from_values(
-                    doc_comment.range.start.line,
-                    1,
-                    doc_comment.range.end.line,
-                    doc_comment.range.end.column)
-                diff.replace(old_range, new_comment.assemble())
+                self.diff_generation(doc_comment, new_comment)
```
Comment (NiklasMM) [inline bears/documentation/DocumentationStyleBear.py]: Related to my [comment](https://github.com/coala/coala/pull/4465#discussion_r128540014) in the other PR: The fact that you store the diff and affected code in a member during this call as a side effect makes it hard for a person reading this code to understand what's going on. I'd prefer it if the results would be returned.

**Comment by Nosferatul** [comment]
@damngamerz you don't need tests for the new functions?

**Comment by damngamerz** [comment]
@Nosferatul Yup We don't need it,It's already there. Im just revamping the structure.

File: bears/documentation/DocumentationStyleBear.py [hunk 3]
```diff
@@ -46,75 +110,29 @@ def run(self, filename, file, language: str,
...
-            parsed = doc_comment.parse()
-            metadata = iter(parsed)
 
-            # Assuming that the first element is always the only main
-            # description.
-            main_description = next(metadata)
+        # Need to initialize them as some parts of extraction and
+        # diff_generation uses them.
+        self.file = file
+        self.filename = filename
```
Comment (NiklasMM) [inline bears/documentation/DocumentationStyleBear.py]: these two member assignments are not needed anymore, right?

File: bears/documentation/DocumentationStyleBear.py [hunk 4]
```diff
@@ -46,75 +110,29 @@ def run(self, filename, file, language: str,
...
-
-                # We need to update old comment positions, as `assemble()`
-                # prepends indentation for first line.
-                old_range = TextRange.from_values(
-                    doc_comment.range.start.line,
-                    1,
-                    doc_comment.range.end.line,
-                    doc_comment.range.end.column)
-                diff.replace(old_range, new_comment.assemble())
+                (affected_code, diffs) = self.generate_diff(
```
Comment (NiklasMM) [inline bears/documentation/DocumentationStyleBear.py]: As I commented in the other PR, accept the plain `Diff` object from the `generate_diff` method and assemble the `{"filename": <Diff>}` object here in this method... makes it clearer what's going on

File: bears/documentation/DocumentationStyleBear.py (hunk 3 above)
Comment (damngamerz) [inline bears/documentation/DocumentationStyleBear.py]: Yup^ amending the changes

File: bears/documentation/DocumentationStyleBear.py (hunk 4 above)
Comment (damngamerz) [inline bears/documentation/DocumentationStyleBear.py]: Yeah we should do this :tada:

**Comment by SanketDG** [comment]
ack 13f515d

File: bears/documentation/DocumentationStyleBear.py [hunk 5]
```diff
@@ -20,6 +19,69 @@ class DocumentationStyleBear(LocalBear):
     CAN_DETECT = {'Documentation'}
     CAN_FIX = {'Documentation'}
 
+    def process_documentation(self, parsed, allow_missing_func_desc: str=False,
+                              indent_size: int=4):
+        """
+        This fixes the parsed documentation comment.
+
+        :param parsed: Contains parsed documentation comment.
```
Comment (SanketDG) [inline bears/documentation/DocumentationStyleBear.py]: put newlines after the param name

**Comment by SanketDG** [comment]
ack d420a70 554b877

**Comment by SanketDG** [comment]
reack 314dfdf

**Comment by SanketDG** [comment]
@rultor merge

**Pull request merged**